### PR TITLE
[WIP] More components

### LINKED ISSUE #14: More reusable components
#13 introduced some stuff, but the component grind continues on. Here's a couple ideas:

- [ ] text area (text area field)
- [ ] radio
- [ ] button variants
- [ ] header
- [ ] input list (more feature rich select)
- [x] success/warning/info messages
- [ ] tag
- [x] more stars
- [ ] popcorn
- [x] icons

### PR DESCRIPTION
**Pull request by scootyboots**
Made the star animation better

Adds info banners and some icons

Screenshot 2024-11-25 at 20 18 57

closes #14

### CODE REVIEW

File: islands/Button.tsx [hunk 1]
```diff
@@ -3,11 +3,29 @@ import { ComponentChildren, JSX } from 'preact'
...
   children,
   type = 'button',
+  fullWidth,
   ...buttonProps
-}: { children: ComponentChildren } & JSX.HTMLAttributes<HTMLButtonElement>) {
+}: {
+  fullWidth?: boolean
+  children: ComponentChildren
+} & JSX.HTMLAttributes<HTMLButtonElement>) {
+  if (fullWidth)
```
Comment (aburd) [inline islands/Button.tsx:12]: I highly suggest this lib
https://www.npmjs.com/package/clsx

File: islands/FormIsland/FormIsland.tsx [hunk 2]
```diff
@@ -6,6 +6,23 @@ import InputField from '../form/InputField.tsx'
 import CheckboxField from '../form/CheckboxField.tsx'
 import Stars from '../stars/Stars.tsx'
 import Divider from '../../components/Divider.tsx'
+import Info from '../../components/Info.tsx'
+import ControlledInfo from '../../components/ControlledInfo.tsx'
+
+import Close from '../../components/icons/close.tsx'
```
Comment (aburd) [inline islands/FormIsland/FormIsland.tsx:12]:
```suggestion
import Close from '../../components/icons/Close.tsx'
```

File: components/ControlledInfo.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import Info, { InfoProps } from '../components/Info.tsx'
+import Close from '../components/icons/close.tsx'
```
Comment (aburd) [inline components/ControlledInfo.tsx:2]:
```suggestion
import Close from '../components/icons/Close.tsx'
```

**Comment by aburd** [comment]
Lookin' pretty dang slick!
I honestly didn't know this app was going to look so pro. I was just thinkin' something quick and dirty but hella good work.

File: islands/Button.tsx (hunk 1 above)
Comment (scootyboots) [reply islands/Button.tsx]: very cool, i'll check it out

File: components/Info.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      </div>
+    )
+  return (
+    <div class="w-full bg-primary-4 border-primary border text-foreground rounded p-5 flex gap-2 justify-between items-center">
+      {icon && <StarIcon width="w-6" />}
+      <div class="w-full">{message}</div>
+      {after}
+    </div>
+  )
+}
```
[the remark this answers is not in the document]
Comment (scootyboots) [reply components/Info.tsx]: yeah i'll tighten this up, dunno why i had it in my head that tailwind doesn't play nice with encapsulating the utility classes inside of variables

this my brain has turned rotten from working with a css-in-js component library for the past few years

File: tailwind.config.ts [hunk 5]
```diff
@@ -6,15 +6,19 @@ export default {
...
         gray: 'rgba(57 64 75)',
         background: 'rgb(24 5 39)',
         'background-low': 'rgb(89 32 111)',
         foreground: 'white',
         'foreground-back': '#cebfcc',
         primary: '#39c1ce',
+        'primary-2': '#23929e',
+        'primary-3': 'rgb(27 110 147)',
+        'primary-4': 'rgb(26 57 91)',
         highlight: 'rgb(177 255 247)',
```
[the remark this answers is not in the document]
Comment (scootyboots) [reply tailwind.config.ts]: yeah i agree the naming isn't great, but the one big advantage with this kind of approach is if in the future we want to support light/dark mode we will be saving ourselves a lot of trouble

File: components/Info.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (aburd) [reply components/Info.tsx]: > think my brain has turned rotten from working with a css-in-js component library for the past few years

Speaking as a recovering css-in-js guy, I pulling for you buddy. There is hope.

File: tailwind.config.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (aburd) [reply tailwind.config.ts]: Ahhh, like so you would just add a 1 or 2 for dark-mode or somethin'. That's pretty cool.

[the remark this answers is not in the document]
Comment (aburd) [reply tailwind.config.ts]: Just FYI, there's also this pkg 
https://www.npmjs.com/package/tw-colors

**Comment by scootyboots** [comment]
conflicts have been resolved, i swear i'll use `deno fmt` like a good boy now

**Pull request merged**